cli/pkg/local/app.go

```diff
@@ -624,6 +632,49 @@ func initLogger(isVerbose bool, logFormat LogFormat) (logger *zap.Logger, cleanu
 	}
 }
 
+func emitSourceTelemetry(ctx context.Context, opts *AppOptions, rt *runtime.Runtime, inst *drivers.Instance) error {
+	controller, err := rt.Controller(ctx, inst.ID)
+	if err != nil {
+		return err
+	}
+	catalog, _, err := rt.Catalog(ctx, inst.ID)
+	if err != nil {
+		return err
+	}
+	resources, err := catalog.FindResources(ctx)
+	if err != nil {
+		return err
+	}
+	var sourceDrivers []string
+	var olapDrivers []string
+	for _, res := range resources {
+		if res.Kind == "rill.runtime.v1.Source" {
+			source, err := controller.Get(ctx, &runtimev1.ResourceName{Kind: res.Kind, Name: res.Name}, false)
+			if err != nil {
+				return err
+			}
+			spec := source.GetSource().Spec
+			sourceDrivers = append(sourceDrivers, spec.SourceConnector)
+			olapDrivers = append(olapDrivers, spec.SinkConnector)
+		}
+	}
```

Comment:
This has a problem, which is that the catalog is often not correctly populated right when the application starts (needs to wait for reconcile).

Instead, it would be better to use the `AnalyzeConnectors` function on the project parser, which is designed to explicitly extract info about the connectors used in the project. You can see the code for the `rill env configure` command for an example.

Note that the `AnalyzeConnectors` function covers everything, not just sources. I think it might be better just to emit the names of all `connectors` and then separately emit the name of the main `OLAPConnector`.